src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -778,8 +747,8 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VSMMetadata,
       annotateSamples(m.getOrElse(_, nullValue), finalType, inserter)
     } else {
       keyTypes match {
-        case Array(TString) =>
-          var r = keyedRDD.map { case (k, v) => (k.asInstanceOf[Row].getAs[String](0), v: Annotation) }
+        case Array(`sSignature`) =>
```

Comment:
I can't see the rest of the code easily but I think you need to fix the error message below